app/controllers/generic_object_definition_controller.rb

```diff
@@ -166,8 +166,12 @@ def custom_buttons_in_set
                        end
     generic_object_definition = find_record_with_rbac(GenericObjectDefinition, params[:generic_object_definition_id])
     unassigned_buttons = generic_object_definition.custom_buttons
-    assigned_buttons.map! { |button| {:name => button.name, :id => button.id} }
-    unassigned_buttons.map! { |button| {:name => button.name, :id => button.id} }
+    unless assigned_buttons.empty?
```

Comment:
This should not be needed... empty array maps will work fine and you'll still get an empty array.

That being said, it's weird to be that we take the result of `generic_object_definition.custom_buttons`, and then modify it...that's bad because custom_buttons is owned by the generic_object_definition and we are changing what it might have cached internally.  Instead, I think the following might be nicer:


```ruby
    unassigned_buttons = generic_object_definition.custom_buttons
    unassigned_buttons = unassigned_buttons.map { |button| {:name => button.name, :id => button.id} }
```

(so remove the `.map!` in favor or `.map`)
this could also be done similarly for the assigned buttons.